### PR TITLE
Update Property Type endpoints to return `Subgraph`

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

This follows on from #1198. It modifies property type methods front to back to use the `Subgraph` type. It also implements some necessary drivebys to unify the `DependencyContext` objects.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203007126736604/1203160580911226/f) _(internal)_
- [Asana task](https://app.asana.com/0/0/1203157172269850/f) _(internal)_

## 🔍 What does this change?

- Changes the `get_property_types_by_query` endpoint on the Graph API to return a `Subgraph`
- Removes permutations of `*Query` and `*DependencyContext` objects in Rust to be replaced by `StructuralQuery` and `DependencyContext` respectively.
  - We're likely to extend the query resolution logic to optionally also return _incoming_ references, as such there's now one unified object.
- Updates the Rust integration tests to account for API changes
- Updates the HTTP tests to account for API changes
- Regenerates the API client
- Updates the various model classes to account for the new `StructuralQuery` object structure
- Removes the `getAllLatest` method from the data type and property type model classes as it is redundant now.
- Updates the property type GraphQL resolvers and query definitions
- Updates the property type block protocol shims

## 📜 Does this require a change to the docs?

It probably will, this will happen as a follow-up (tracked in this [asana task](https://app.asana.com/0/0/1203160372638987/f) _(internal)_)

## ⚠️ Known issues

- When using the `DependencyContext` object, we're now passing around references inside of the mutable hashmaps we're iterating over. This caused borrowing errors and as such some `clone`s have been added. I view removal of these as a micro-optimisation
- We don't have a nice place in the Rust code at the moment for objects which relate to _both_ the Knowledge Graph and the Ontology. Such objects are currently stored in `subgraph.rs`, this will be fixed in a follow-up, and will relate to the work in #1197 
- I noticed an inconsistency with our visibilities in the `Persisted*` structs. This will be fixed in a follow-up https://github.com/hashintel/hash/pull/1206 (TODOs were added)

## 🐾 Next steps

- Update `DependencyMap` logic to track relationships between elements
- Migrate `EntityType` endpoints to use `Subgraph`
- Migrate `LinkType` endpoints to use `Subgraph`
- Migrate `Link` endpoints to use `Subgraph`
- Migrate `Entity` endpoints to use `Subgraph`
- Remove any code that has been made redundant as a result of this work
- Update documentation

## 🛡 What tests cover this?

This was a full stack change, so all the tests of HASH should be checked. These include the ones outlined in the Graph README, as well as workspace integration tests, and manual testing of the application.

## ❓ How to test this?

Follow the READMEs, and look at CI

### CODE REVIEW

**Review by TimDiekmann — COMMENTED** [review]
Great work! The Rust side of the changes looks good 🎉

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx [hunk 1]
```diff
@@ -16,13 +16,25 @@ export const useRemotePropertyTypes = () => {
...
           Object.fromEntries(
-            data.data.results.map(
-              (result) =>
-                [
-                  mustBeVersionedUri(result.propertyTypeId),
-                  result.propertyType,
-                ] as const,
-            ),
+            data.data.roots.map((propertyTypeId) => {
+              const propertyTypeVertex = data.data!.vertices[propertyTypeId!];
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx:20]: Ahah, good spot! I thought that was about the array potentially being empty. My mistake!

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx [hunk 2]
```diff
@@ -16,13 +16,25 @@ export const useRemotePropertyTypes = () => {
...
         setPropertyTypes(
           Object.fromEntries(
-            data.data.results.map(
-              (result) =>
-                [
-                  mustBeVersionedUri(result.propertyTypeId),
-                  result.propertyType,
-                ] as const,
-            ),
+            data.data.roots.map((propertyTypeId) => {
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx]: That also removed another assertion, thanks

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx]: Have addressed

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx [hunk 3]
```diff
@@ -16,13 +16,25 @@ export const useRemotePropertyTypes = () => {
...
+                );
+              }
+              if (propertyTypeVertex.kind !== "PROPERTY_TYPE") {
+                throw new Error(
+                  `expected property type but got ${propertyTypeVertex.kind}`,
+                );
+              }
+
+              return [
+                mustBeVersionedUri(propertyTypeId!),
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-property-types.tsx]: Have addressed

**Review by benwerner01 — APPROVED** [review]
Looks good to me 👍

**Pull request merged**